<a href="https://colab.research.google.com/github/akshatchaturvedi28/Collab-TensorFlow/blob/master/begineer.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [0]:
from __future__ import absolute_import, division, print_function, unicode_literals

# install TensorFlow
!pip install tensorflow==2.0.0

import tensorflow as tf

Loading the MNIST DATASET

In [0]:
mnist = tf.keras.datasets.mnist


In [0]:
(x_train, y_train), (x_test, y_test) = mnist.load_data()
x_train, x_test = x_train/255.0, x_test/255.0

11493376/11490434 [==============================] - 0s 0us/step


Build the tf.keras.Sequential model by stacking layers. Choose an optimizer and a loss function for training:

In [0]:
model = tf.keras.models.Sequential([
    tf.keras.layers.Flatten(input_shape=(28, 28)),
    tf.keras.layers.Dense(128, activation='relu'),
    tf.keras.layers.Dropout(0.2),
    tf.keras.layers.Dense(10, activation='softmax')
])

model.compile(optimizer='adam', 
             loss='sparse_categorical_crossentropy',
             metrics=['accuracy'])

Train and Evaluate the model

In [0]:
model.fit(x_train, y_train, epochs=5)

model.evaluate(x_test, y_test)

Train on 60000 samples
Epoch 1/5
60000/60000 [==============================] - 8s 129us/sample - loss: 0.2964 - accuracy: 0.9139
Epoch 2/5
60000/60000 [==============================] - 6s 98us/sample - loss: 0.1417 - accuracy: 0.9570
Epoch 3/5
60000/60000 [==============================] - 6s 96us/sample - loss: 0.1068 - accuracy: 0.9677
Epoch 4/5
60000/60000 [==============================] - 6s 97us/sample - loss: 0.0868 - accuracy: 0.9726
Epoch 5/5
60000/60000 [==============================] - 6s 97us/sample - loss: 0.0739 - accuracy: 0.9768
10000/1 [=====================================================================================================================================================================================================================================================================================================================================================================================================================================================

[0.07229477977477945, 0.9782]

The image classifier is now trained to ~98% accuracy on this dataset.